{"version":3,"sources":["../../../src/server/after/after.ts"],"sourcesContent":["import { workAsyncStorage } from '../app-render/work-async-storage.external'\n\nexport type AfterTask<T = unknown> = Promise<T> | AfterCallback<T>\nexport type AfterCallback<T = unknown> = () => T | Promise<T>\n\n/**\n * This function allows you to schedule callbacks to be executed after the current request finishes.\n */\nexport function unstable_after<T>(task: AfterTask<T>): void {\n  const workStore = workAsyncStorage.getStore()\n\n  if (!workStore) {\n    // TODO(after): the linked docs page talks about *dynamic* APIs, which unstable_after soon won't be anymore\n    throw new Error(\n      '`unstable_after` was called outside a request scope. Read more: https://nextjs.org/docs/messages/next-dynamic-api-wrong-context'\n    )\n  }\n\n  const { afterContext } = workStore\n  if (!afterContext) {\n    throw new Error(\n      '`unstable_after` must be explicitly enabled by setting `experimental.after: true` in your next.config.js.'\n    )\n  }\n\n  return afterContext.after(task)\n}\n"],"names":["workAsyncStorage","unstable_after","task","workStore","getStore","Error","afterContext","after"],"mappings":"AAAA,SAASA,gBAAgB,QAAQ,4CAA2C;AAK5E;;CAEC,GACD,OAAO,SAASC,eAAkBC,IAAkB;IAClD,MAAMC,YAAYH,iBAAiBI,QAAQ;IAE3C,IAAI,CAACD,WAAW;QACd,2GAA2G;QAC3G,MAAM,IAAIE,MACR;IAEJ;IAEA,MAAM,EAAEC,YAAY,EAAE,GAAGH;IACzB,IAAI,CAACG,cAAc;QACjB,MAAM,IAAID,MACR;IAEJ;IAEA,OAAOC,aAAaC,KAAK,CAACL;AAC5B"}